rts/lra/lra-coordinator/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java

```diff
@@ -790,15 +832,34 @@ public boolean afterLRANotification() {
 
                 while (listeners.hasNext()) {
                     URI uri = listeners.next();
+                    Response response = null;
+
+                    try {
+                        Future<Response> asyncResponse = client.target(uri)
+                                .request()
+                                .header(LRA.LRA_HTTP_ENDED_CONTEXT_HEADER, id)
+                                .async()
+                                .put(Entity.text(status.name()));
+
+                        response = asyncResponse.get(PARTICIPANT_TIMEOUT, TimeUnit.SECONDS);
 
-                    Response response = client.target(uri)
-                            .request()
-                            .header(LRA.LRA_HTTP_ENDED_CONTEXT_HEADER, id)
-                            .put(Entity.text(status.name()));
-                    if (response.getStatus() == 200) {
-                        listeners.remove();
-                    } else {
+                        if (response.getStatus() == 200) {
+                            listeners.remove();
+                        } else {
+                            notifiedAll = false;
+                        }
+
+                        response = null;
+                    } catch (WebApplicationException | InterruptedException | ExecutionException | TimeoutException e) {
                         notifiedAll = false;
+
+                        if (LRALogger.logger.isInfoEnabled()) {
+                            LRALogger.logger.infof("Could not notify AfterLRA listener at %s (%s)", uri, e.getMessage());
```

Comment:
I wonder in general here, why the caught exception is printed as the info message. Should not be a warning when it's something exceptional?

Reply:
The participant that is being notified may not be running at the moment and it will be retried later. This is not an exceptional condition is a distributed system. I can log it at debug level if you prefer but I thought info would be more helpful.

Reply:
I see but I think the unavailable resource is exceptional. It's the same when JTA recovery tries to run 'rollback'. It's not expected behaviour and thus is for WARN.
It could be about the log level interpretation but a caught exception is an exception even in the distributed system.